test.py

```diff
@@ -0,0 +1,162 @@
+
+class RegulatoryElement(BaseModel):
+    type: str
+    value_id: str
+    label: StrictStr
+
+    @validator('type')
+    def valid_reg_type(cls, v):
+        assert v == 'promoter' or v == 'enhancer', 'type must be either promoter or enhancer'
+        return v
+
+    @validator('value_id')
+    def valid_id(cls, v):
+        assert v.find(':') != -1 and v.find(' ') == -1, 'id must be a CURIE'
+        return v
+
+class Fusion(BaseModel):
+    r_frame_preserved: StrictBool
+    regulatory_elements: List[RegulatoryElement]
+    protein_domains: List[CriticalDomain]
+    transcript_components = []
+    causative_event: Event
+
+    def make_json(self):
+        return(json.dumps(self, default=lambda o: o.__dict__, indent=4))
+
+def main():
+
+    try:
+        gen1 = Gene(value_id = 'hgnc:12012', label = 'TPM3')
+        gen2 = Gene(value_id = 'hgnc:8031', label = 'NTRK1')
+        ex1 = GenomicRegion(value = SequenceRegion(type = 'SequenceLocation', seq_id = 'ncbi:NC_000001.11',  start =  154170399, end = 1556346346))
+        ex2 = GenomicRegion(value = SequenceRegion(type = 'SequenceLocation', seq_id = 'ncbi:NC_000001.13',  start =  154170401, end = 1556346348))
+        linker_seq = Linker(linker_sequence = 'ATTATTA')
+        cause_eve = Event(event_type = 'rearrangement')
+        domain = CriticalDomain(status = 'preserved', name = 'tyrosine kinase catalytic domain', id = 'interpro:IPR020635',
+                         gene = gen2)
+        reg1 = RegulatoryElement(type = 'promoter', value_id = 'hgnc:100', label = 'ASIC1')
+        reg2 = RegulatoryElement(type = 'enhancer', value_id = 'hgnc:200', label = 'G1')
+        ur1 = SequenceRegion(type = 'SequenceLocation', seq_id = 'ncbi:NC_000001.11', start = 1000, end = 5000)
+        c1 = UnknownComponent(region = ur1)
+        c2 = UnknownComponent()
+        ur2 = ChromosomeRegion(type = 'ChromosomeLocation', chr = '12', start = 'p12.1', end = 'p12.16')
+        c3 = UnknownComponent(region = ur2)
+        tr1 = TranscriptRegion(transcript = 'NM:152263.3', exon_start = 1, exon_start_offset = -9,
+                                  exon_end = 8, exon_end_offset = 7, gene = gen1, component_genomic_region = ex1)
+        tr2 = TranscriptRegion(transcript = 'NM:002529.3', exon_start = 10, exon_start_offset = -8,
+                                  exon_end = 17, exon_end_offset = 8, gene = gen2, component_genomic_region = ex2)
+        molecular_fusion = Fusion(r_frame_preserved = True, regulatory_elements = [reg1, reg2], protein_domains = [domain],
+                                  transcript_components = [c1, c2, c3, tr1, linker_seq, tr2], causative_event = cause_eve,)
+        print(molecular_fusion.json())
+    except ValidationError as f:
+        print(f)
+
+    filename = 'example_gene_fusion_class.json'
+    with open(filename, 'w') as files:
+        files.write(molecular_fusion.make_json())
+    files.close()
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
I don't think this should go here. Maybe create a directory called analysis and put it there? You should consider creating tests (we use pytest)!

Reply:
Yes, my plan is to copy this over into another file for testing.